addons/fleet/models/fleet_vehicle.py

```diff
@@ -200,36 +198,33 @@ def _compute_contract_reminder(self):
         for record in self:
             overdue = False
             due_soon = False
-            total = 0
-            name = ''
             state = ''
-            for element in record.log_contracts:
-                if element.state in ('open', 'expired') and element.expiration_date:
-                    current_date_str = fields.Date.context_today(record)
-                    due_time_str = element.expiration_date
-                    current_date = fields.Date.from_string(current_date_str)
-                    due_time = fields.Date.from_string(due_time_str)
-                    diff_time = (due_time - current_date).days
-                    if diff_time < 0:
-                        overdue = True
-                        total += 1
-                    if diff_time < delay_alert_contract:
-                        due_soon = True
-                        total += 1
-                    if overdue or due_soon:
-                        log_contract = self.env['fleet.vehicle.log.contract'].search([
-                            ('vehicle_id', '=', record.id),
-                            ('state', 'in', ('open', 'expired'))
-                            ], limit=1, order='expiration_date asc')
-                        if log_contract:
-                            # we display only the name of the oldest overdue/due soon contract
-                            name = log_contract.name
-                            state = log_contract.state
+            log_contract = self.env['fleet.vehicle.log.contract'].search(
+                [
+                    ('vehicle_id', '=', record.id),
+                    ('state', 'in', ('futur', 'open', 'expired')),
+                ],
+                limit=1, order='expiration_date desc')
+
+            if log_contract.expiration_date:
+                current_date = fields.Date.context_today(record)
+                due_time = log_contract.expiration_date
+                diff_time = (due_time - current_date).days
+                overdue = diff_time < 0
+                due_soon = (diff_time < delay_alert_contract) and not overdue
+                state = log_contract.state
+
+            if state == 'futur':
+                last_active_contract_state = self.env['fleet.vehicle.log.contract'].search(
+                    [
+                        ('vehicle_id', '=', record.id),
+                        ('state', 'in', ('open', 'expired')),
+                    ],
+                    limit=1, order='expiration_date desc').state
+                last_active_contract_state and (state := last_active_contract_state)
```

Comment:
This is in any case not very readable. Using the ':=' operator is rarely the most readable thing to do in my opinion. Also what you're doing here is basically
```python
if last_active_contract_state:
    state = last_active_contract_state
```
which is way more readable (even though you're using an extra line).